ape_solidity/compiler.py

```diff
@@ -413,17 +426,18 @@ def compile(
                                 ct for ct in contract_types if ct.name != contract_name
                             ]
 
-                    contract_type["contractName"] = contract_name
-                    contract_type["sourceId"] = str(
+                    ct_data["contractName"] = contract_name
+                    ct_data["sourceId"] = str(
                         get_relative_path(base_path / contract_path, base_path)
                     )
-                    contract_type["deploymentBytecode"] = {"bytecode": deployment_bytecode}
-                    contract_type["runtimeBytecode"] = {"bytecode": runtime_bytecode}
-                    contract_type["userdoc"] = load_dict(contract_type["userdoc"])
-                    contract_type["devdoc"] = load_dict(contract_type["devdoc"])
-                    contract_type["sourcemap"] = evm_data["bytecode"]["sourceMap"]
-                    contract_type_obj = ContractType.parse_obj(contract_type)
-                    contract_types.append(contract_type_obj)
+                    ct_data["deploymentBytecode"] = {"bytecode": deployment_bytecode}
+                    ct_data["runtimeBytecode"] = {"bytecode": runtime_bytecode}
+                    ct_data["userdoc"] = load_dict(ct_data["userdoc"])
+                    ct_data["devdoc"] = load_dict(ct_data["devdoc"])
+                    ct_data["sourcemap"] = evm_data["bytecode"]["sourceMap"]
+                    ct_data["ast"] = ast
```

Comment:
would be nice to be able to filter out AST nodes that are specific to this contract, but not sure that is a very easy task.

(currently if 2 contracts are in same file, they have exact same AST structure)